### PR TITLE
[airtunes/linux] - fix airtunes/shairplay not working on linux

### PR DESCRIPTION
**Pull request by Memphiz**
## Description
With latest iOS clients airtunes did not work on linux (tested on ubuntu 18.04 here) - it just didn't start the music stream.

This PR fixes it by:
1. bumping lib shairplay to latest master of upstream
2. fixing PaPlayer which ignored forced current and total times when there was no stream yet
3. Ensure that AirTunes server remembers the progress values (current time / total time) from the lib shairplay callback and forward those to the player once the player is available.

This fixes also wrong track duration after the first start of airplay (seeking in the iOS client fixed it before), by delaying the progress information until the player our available.

## Motivation and Context
AirTunes non-working on linux

## How Has This Been Tested?
Tested on linux ubuntu 18.04 and MacOS High Sierra with iPhone running 12.1.1.

## Types of change

- [X] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by Memphiz** [comment]
@wsnipex fyi for the bumped lib
@FernetMenta for the paplayer change
@MartijnKaijser for deciding if this can go to Leia or not.

**Comment by Memphiz** [comment]
For completeness. I also tested on windows 10. There is no issue with the old lib. Technically there was no issue with the old lib on mac OS too. Only linux didn't work. But i figured if i bump depends i bump it for all depends platforms to prevent any ifdefery in there. So Windows does not need a lib bump (but also benefits from the paplayer fix ...)

**Comment by MartijnKaijser** [comment]
I'm ok with it

**Review by mkreisl — COMMENTED** [review]
xbmc/platform/linux/network/ZeroconfAvahi.cpp line 366
return false must be inside of {}

File: xbmc/platform/linux/network/ZeroconfAvahi.cpp [hunk 1]
```diff
@@ -355,9 +355,14 @@ bool CZeroconfAvahi::createClient()
...
+
+    int error = 0;
     mp_client = avahi_client_new(avahi_threaded_poll_get(mp_poll),
-                                 AVAHI_CLIENT_NO_FAIL, &clientCallback,this,0);
+                                 AVAHI_CLIENT_NO_FAIL, &clientCallback,this,&error);
     if (!mp_client)
+    {
+      CLog::Log(LOGERROR, "CZeroconfAvahi::createClient() failed with %d", error);
+    }
       return false;
```
Comment (pkerling) [inline xbmc/platform/linux/network/ZeroconfAvahi.cpp]: brace/indent mismatch

**Review by yol — DISMISSED** [review]

**Comment by pkerling** [comment]
@mkreisl you can add comments to the code :-)

**Comment by mkreisl** [comment]
> @mkreisl you can add comments to the code :-)

Was trying it, but it seems I'm too stupid for that

**Comment by Memphiz** [comment]
ha good catch - and that’s why I hate missing braces - I tend to be to stupid to add them at the right spots ;)

File: xbmc/platform/linux/network/ZeroconfAvahi.cpp (hunk 1 above)
Comment (Memphiz) [reply xbmc/platform/linux/network/ZeroconfAvahi.cpp]: fixed

**Comment by pkerling** [comment]
> and that’s why I hate missing braces

I'm also a strong proponent of having braces everywhere, but with the coding style Kodi has (every curly brace on new line) that does tend to take up a lot of vertical space

**Review by FernetMenta — APPROVED** [review]

**Comment by Memphiz** [comment]
Updated with a compile fix which i also proposed upstream as PR. Lets see what jenkins thinks about it now :)

File: xbmc/cores/paplayer/PAPlayer.cpp [hunk 2]
```diff
@@ -956,26 +960,32 @@ int64_t PAPlayer::GetTimeInternal()
...
 }
 
-void PAPlayer::SetTotalTimeInternal(int64_t time)
+bool PAPlayer::SetTotalTimeInternal(int64_t time)
 {
   CSingleLock lock(m_streamsLock);
   if (!m_currentStream)
-    return;
+  {
+    return false;;
```
Comment (enen92) [inline xbmc/cores/paplayer/PAPlayer.cpp]: ^^ two semi-colons

**Comment by Memphiz** [comment]
fixed the double semicolons ... jenkins build error for iOS-arm64 was unrelated. But it seems like master iOS ARM 64 is broken ...

**Comment by wsnipex** [comment]
new shairplay version pushed to the nightly PPA, builds and runs in general, but I cannot test airplay support.

**Comment by chewitt** [comment]
I'm streaming SomaFM from an iPhone 8 (iOS 12.2.1) to an Amlogic mainline LE device with this PR and the corresponding LE shairport package change in https://github.com/LibreELEC/LibreELEC.tv/pull/3179 and all works fine.

**Comment by Memphiz** [comment]
Thx for the test :)

**Comment by MartijnKaijser** [comment]
@Memphiz merge ready?

**Comment by Memphiz** [comment]
Yep

**Pull request merged**